src/components/StudentsTable.vue

```diff
@@ -60,15 +103,59 @@ const onCellEditingStopped = (params: CellEditingStoppedEvent) => {
   if (someValueChanged.value) canSubmit.value = true
 }
 
+const onSelectionChanged = () => {
+  selectedRows.value = studentsTableApi.value?.getSelectedRows() ?? []
+}
+
+const getRowId = (params: GetRowIdParams) => params.data.id.value
+
+const ensureRowsDataIsValid = (rows: StudentWithMetadata[]) =>
+  rows.every((student) => Object.values(student).every(({ isValidated }) => isValidated))
+
 const updateStudents = () => {
   studentsStore.updateStudents({
+    newStudents: addedRows.value,
     onError: (errorMessage: string) => alert(errorMessage),
     onSuccess: () => {
       canSubmit.value = false
       someValueChanged.value = false
+      addedRows.value = []
     }
   })
 }
+
+const addRow = () => {
+  const emptyStudent = {
+    id: { value: crypto.randomUUID(), isValidated: true },
+    name: { isValidated: false },
+    lastName: { isValidated: false },
+    birthDate: { isValidated: false },
+    finalGrade: { isValidated: false },
+    hobbies: { value: [], isValidated: true }
+  }
+
+  studentsTableApi.value?.applyTransaction({
+    add: [emptyStudent]
+  })
+
+  addedRows.value.push(emptyStudent)
+  studentsStore.students?.push(emptyStudent)
```

Comment:
This approach will punish you sooner or later. It's one of the biggest issues while working with ag grid.
Some time after you add more and more stuff you will experience grid/store desync.
Tbh we still don't know what is the best approach, but you should definitely pick smth until someone smarter figures it out:
1st - ag grid is the main hero, and store updates itself after smth is changed in table.
2nd - store is the main hero, and ag grid changes data when store is updated